tools/localnet/design.md

```diff
@@ -0,0 +1,93 @@
+# LOCALNET
+
+`localnet` is a tool allowing us to set up the entire local ecosystem required to build and test new features.
+
+## Motivation
+
+By providing new features and integrating new blockchains, more and more funds are collected in our ecosystem. It means that
+responsibility of dev team grows every day. That's why we need good tools helping us to build our confidence that new
+feature works well before pull request is even created by the author.
+
+Before IBC era, testing new features locally was pretty easy. The only required component was a single `sifnode`.
+Now, after releasing IBC it is much more complicated, and many interconnected components are needed.
+For ease of use it seems reasonable to create a tool starting and managing all of them.
+
+## Requirements
+
+To be able to play with all th features of `sifnode` (including IBC) we need these components:
+- at least two `sifnodes` running two separate chains
+- relayer between them
+- local eth testnet
+- peggy between eth and sifchain
+- block explorer for every chain
+
+
+Every time we add new supported blockchain `localnet` is extended by adding testnet, relayer  and block explorer.
```

Comment:
we'll probably want to create some parameterized config, to set which blockchain(s) the localnet should include, otherwise it could get unwieldy if we have 5+. Also wondering how we address block history for a given chain?

Reply:
> we'll probably want to create some parameterized config, to set which blockchain(s) the localnet should include, otherwise it could get unwieldy if we have 5+

yes, I thought about it, we may provide some CLI flags: `--with-blockchain-ethereum`, `--with-blockchain-<blockchain-name>`

> Also wondering how we address block history for a given chain?

What do you mean? Every blockchain is a separate "database" stored in separate directory and accessed by querying its node